components/eam/src/chemistry/modal_aero/aero_model.F90

```diff
@@ -1631,6 +1636,59 @@ subroutine aero_model_wetdep(dt, dlf, dlf2, cmfmc2, state,                    &!
        rsscavt_cn_sv(:,:) = 0.0_r8
     endif
 
+    if (convproc_do_aer) then
+
+       call pbuf_get_field(pbuf, icwmrdp_idx,     icwmrdp )
+       call pbuf_get_field(pbuf, icwmrsh_idx,     icwmrsh )
+       call pbuf_get_field(pbuf, sh_frac_idx,     sh_frac )
+       call pbuf_get_field(pbuf, dp_frac_idx,     dp_frac )
+
+       call t_startf('ma_convproc')
+       call ma_convproc_intr( state, ptend, pbuf, dt,                   &
+            dp_frac, icwmrdp, rprddp, evapcdp,                          &
+            sh_frac, icwmrsh, rprdsh, evapcsh,                          &
+            dlf, dlf2, cmfmc2, sh_e_ed_ratio,                           &
+            nsrflx_mzaer2cnvpr, qsrflx_mzaer2cnvpr, aerdepwetis,        &
+            mu, md, du, eu, ed, dp, dsubcld, jt, maxg, ideep, lengath,  &
+            species_class, mam_prevap_resusp_optaa,                     &
+            history_aero_prevap_resusp, &
+            dcondt_resusp3d=dcondt_resusp3d,wuc=wuc)
+    do m = 1, ntot_amode ! main loop over aerosol modes
+     do lphase = strt_loop,end_loop, stride_loop
+      ! loop over interstitial (1) and cloud-borne (2) forms
+      do lspec = 0, nspec_amode(m) !+1 ! loop over number + chem constituents + water
+       if (lspec == 0) then ! number
+        if (lphase == 1) then
+         mm = numptr_amode(m)
+        else
+         mm = numptrcw_amode(m)
+        endif
+       else if (lspec <= nspec_amode(m)) then ! non-water mass
+        if (lphase == 1) then
+         mm = lmassptr_amode(lspec,m)
+        else
+         mm = lmassptrcw_amode(lspec,m)
+        endif
+       endif
+       if (lphase .eq. 2) then
+        fldcw => qqcw_get_field(pbuf, mm,lchnk)
+        fldcw(:,:) = fldcw(:,:) + dcondt_resusp3d(mm+pcnst,:,:) !*dt
+!The dcondt_resusp3d is detrained aerosol AMOUNT (i.e., both mass
+!and number for four modals).
+!For dcondt_resusp3d(idx,:,:), idx=16,...,40 -> interstial aerosols
+!                              idx=56,...,80 -> cloud-borne aerosols
+! Currently, we assume that no aerosol resuspension occurrs during convective
+! cloud water detrainment. So, 100% of cloud-borne aerosols will be added to
+! stratiform cloud-borne aerosols and detrained interstial aerosol amount is
+! zero.
```

Comment:
Indent comments to match the code. Changing the indentation hurts readability.
Also, should "modals" be "modes"?

Reply:
> Indent comments to match the code. Changing the indentation hurts readability. Also, should "modals" be "modes"?

Thanks for your comments. I revised the indentation and correct the typo "modals"-->"modes".